File: client-android/lib/src/test/kotlin/stasis/test/client_android/lib/discovery/ServiceApiEndpointSpec.kt
package stasis.test.client_android.lib.discovery

import io.kotest.core.spec.style.WordSpec
import io.kotest.matchers.shouldBe
import stasis.client_android.lib.discovery.ServiceApiEndpoint
import stasis.client_android.lib.model.core.networking.EndpointAddress

class ServiceApiEndpointSpec : WordSpec({
    "A ServiceApiEndpoint" should {
        "support providing an ID" {
            ServiceApiEndpoint.Api(uri = "test-uri").id shouldBe ("api__test-uri")

            ServiceApiEndpoint.Core(
                address = EndpointAddress.HttpEndpointAddress(uri = "test-uri")
            ).id shouldBe ("core_http__test-uri")

            ServiceApiEndpoint.Core(
                address = EndpointAddress.GrpcEndpointAddress(host = "test-host", port = 1234, tlsEnabled = false)
            ).id shouldBe ("core_grpc__test-host:1234")
        }
    }
})

File: core/src/test/scala/stasis/test/specs/unit/core/discovery/ServiceApiEndpointSpec.scala
package stasis.test.specs.unit.core.discovery

import stasis.core.discovery.ServiceApiEndpoint
import stasis.core.networking.EndpointAddress
import stasis.core.networking.grpc.GrpcEndpointAddress
import stasis.core.networking.http.HttpEndpointAddress
import io.github.sndnv.layers.testing.UnitSpec

class ServiceApiEndpointSpec extends UnitSpec {
  "A ServiceApiEndpoint" should "support providing an ID" in {
    ServiceApiEndpoint.Api(uri = "test-uri").id should be("api__test-uri")

    ServiceApiEndpoint
      .Core(
        address = HttpEndpointAddress(uri = "test-uri")
      )
      .id should be("core_http__test-uri")

    ServiceApiEndpoint
      .Core(
        address = GrpcEndpointAddress(host = "test-host", port = 1234, tlsEnabled = false)
      )
      .id should be("core_grpc__test-host:1234")

    an[IllegalArgumentException] should be thrownBy ServiceApiEndpoint.Core(address = new EndpointAddress {}).id

    ServiceApiEndpoint.Discovery(uri = "test-uri").id should be("discovery__test-uri")
  }
}
